server/src/main/java/org/apache/iotdb/db/engine/StorageEngine.java

```diff
@@ -122,6 +105,7 @@ public class StorageEngine implements IService {
   private static OperationSyncProducer operationSyncProducer;
   private static OperationSyncDDLProtector operationSyncDDLProtector;
   private static OperationSyncLogService operationSyncDDLLogService;
+  private static boolean isSecondaryLife;
```

Comment:
This field should better to be set to AtomicBoolean or volatile for thread-safety